Okta/OktaAuth/OktaAppAuth.swift

```diff
@@ -10,98 +10,105 @@
+        didSet {
+            tokenManager?.writeToSecureStorage()
+        }
     }
-}
-
-public var isAuthenticated: Bool {
-    return tokenManager?.accessToken != nil
-}
 
-public func signInWithBrowser(from presenter: UIViewController, callback: @escaping ((OktaTokenManager?, OktaError?) -> Void)) {
-    guard let configuration = configuration else {
-        callback(nil, OktaError.notConfigured)
-        return
+    public var isAuthenticated: Bool {
+        return tokenManager?.accessToken != nil
     }
     
-    SignInTask(presenter: presenter, config: configuration, oktaAPI: OktaRestApi())
-    .run { authState, error in
-        guard let authState = authState else {
-            callback(nil, error)
-            return
+    // Holds the browser session
+    var currentUserSessionTask: UserSessionTask?
+    
+    public init?(configuration: OktaAuthConfig? = nil) {
+        guard let config = configuration ?? (try? OktaAuthConfig.default()) else {
+            return nil
         }
         
-        let tokenManager = OktaTokenManager(authState: authState)
-        OktaAuth.tokenManager = tokenManager
-        callback(tokenManager, nil)
-    }
-}
-
-public func signOutOfOkta(from presenter: UIViewController, callback: @escaping ((OktaError?) -> Void)) {
-    guard let configuration = configuration else {
-        callback(OktaError.notConfigured)
-        return
+        self.configuration = config
     }
 
-    SignOutTask(presenter: presenter, config: configuration, oktaAPI: OktaRestApi())
-    .run { _, error in callback(error) }
-}
+    public func signInWithBrowser(from presenter: UIViewController, callback: @escaping ((OktaTokenManager?, OktaError?) -> Void)) {
+        let signInTask = SignInTask(presenter: presenter, config: configuration, oktaAPI: OktaRestApi())
+        currentUserSessionTask = signInTask
 
-public func authenticate(withSessionToken sessionToken: String, callback: @escaping ((OktaTokenManager?, OktaError?) -> Void)) {
-    guard let configuration = configuration else {
-        callback(nil, OktaError.notConfigured)
-        return
+        signInTask.run { [weak self] authState, error in
+            defer { self?.currentUserSessionTask = nil }
+        
+            guard let authState = authState else {
+                callback(nil, error)
+                return
+            }
+            
+            let tokenManager = OktaTokenManager(authState: authState)
+            self?.tokenManager = tokenManager
+            callback(tokenManager, nil)
+        }
     }
 
-    AuthenticateTask(sessionToken: sessionToken, config: configuration, oktaAPI: OktaRestApi())
-    .run { authState, error in
-        guard let authState = authState else {
-            callback(nil, error)
+    public func signOutOfOkta(from presenter: UIViewController, callback: @escaping ((OktaError?) -> Void)) {
+        guard let idToken = tokenManager?.authState.lastTokenResponse?.idToken else {
```

Comment:
Are we dropping the helper function for `idToken` and `accessToken`?

Mapping `tokenManager?.idToken` to `tokenManager?.authState.lastTokenResponse?.idToken` is a huge DX improvement.

Reply:
Probably we can introduce additional overloaded function that accepts OktaTokenManager instance as a parameter.
The idea behind is the following use case: I can deserialize tokenManager that was the result of authentication with non-Okta service. So  I used my own code for storing/retrieving data from keychain. Since tokenManager setter is internal I can't set proper instance to this property and therefore can't use signOutOfOkta API

Reply:
@anastasiiaiurok-okta - Now I remember.

Please add a code comment explaining why we're referencing a cached id token, instead of going through the helper.